packages/back-end/types/datasource.d.ts

```diff
@@ -153,6 +152,8 @@ export type DataSourceSettings = {
   experimentDimensions?: string[];
   notebookRunQuery?: string;
   informationSchemaId?: string;
+  // Some event trackers (currently only Amplitude) require a projectId to support auto metrics
+  projectId?: string;
```

Comment:
@jdorn Initially, I did go down the route of adding the `Project Id` to the second page of the `NewDataSourceForm` like we do for Matomo's `siteId`, but in doing so, I saw that there was no way for a user to ever edit this field after they create the data source, so I ultimately opted to add it to the first page.

I've reverted this back to use the `schemaOptions` like we do for Matomo.

From what I can tell, up until now, we haven't ever needed to allow a user to edit these `schemaOptions` because they were only used when initially generating a new data source (to do things like generate the default queries). However, in this case, the `projectId` will be used to generate the default metric query, and auto generated metric queries. So if a user makes a typo, all of their initial metric queries will be incorrect, and they won't be able to use the auto metrics feature at all.

If there is a place to edit this that I just haven't found, let me know. 

Otherwise, if you like the updated pattern, I can explore how we can allow users to edit this info via the `DataSourceForm`.

Reply:
@jdorn I just pushed up [this commit](https://github.com/growthbook/growthbook/pull/1522/commits/b02f907e1c2faefe9a39ec59fa4a7870021d0512) that introduces a new component `EditSchemaOptions` to the `DataSourceForm` component. It's set up similar to how we handle `ConnectionSettings`. Currently, it's only provisioned for `Amplitude`, but can be expanded as necessary.

With this, a user can now edit their Amplitude `projectId` - allowing orgs with existing Amplitude data sources to use auto metrics, and supporting the use case where an org entered the projectId incorrectly, or they need to change the projectId.